### PR TITLE
Added ConvertTo-Markdown method

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/10273

## Description
Added ConvertTo-Markdown method

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by glicht** [comment]
is this doing what `tableToMarkdown` does? If so name it `TableToMarkdown`. We are trying to use the same names as in python for utility functions. We can have an alias of `ConvertTo-Markdown` for completeness.

**Comment by kirbles19** [comment]
@hod-alpert Done

**Comment by hod-alpert** [comment]
@glicht Renamed the function

File: Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1 [hunk 1]
```diff
@@ -387,3 +387,72 @@ function ReturnOutputs([string]$ReadableOutput, [object]$Outputs, [object]$RawRe
...
+            }
+            $result += $separator -join ' | '
+            $result += "`n"
+
+            # Writing the values
+            ForEach ($item in $items)
+            {
+                $values = @()
+                $item.PSObject.Properties | ForEach-Object {
+                    $values += $_.Value
```
Comment (glicht) [inline Packs/Base/Scripts/CommonServerPowerShell/CommonServerPowerShell.ps1:445]: I think we need to apply escaping to chars that might break the MD. See stringEscapeMD in cmmonserverpython.

**Review by glicht — CHANGES_REQUESTED** [review]

**Comment by hod-alpert** [comment]
@glicht Added support for escaping characters and for nested objects

**Comment by glicht** [comment]
What about `name` argument for the table as we have in CommonServerPython?

**Comment by hod-alpert** [comment]
@glicht `Name` is already supported.
Added support for hash-tables and non list objects

**Comment by glicht** [comment]
@hod-alpert it is looking better. I would just try to handle nested objects better. I would avoid serializing to json but instead simply use the string representation. Try looking at Format-Table and the output they provide. 

For example output from format-table for `$psversiontable`:
image

Output from tableToMarkdown:
image

**Comment by hod-alpert** [comment]
No luck with `Format-Table`.
However, `Out-String` does the trick IMO.

But with big nested objects it becomes less and less readable.

With Json serialization it looks like this:

| PSVersion | PSEdition | GitCommitId | OS | Platform | PSCompatibleVersions | PSRemotingProtocolVersion | SerializationVersion | WSManStackVersion
| --- | --- | --- | --- | --- | --- | --- | --- | ---
| \{<br>  "Major": 6,<br>  "Minor": 2,<br>  "Patch": 3,<br>  "PreReleaseLabel": null,<br>  "BuildLabel": null<br>\} | Core | Unix | \{<br>  "Major": 1,<br>  "Minor": 1,<br>  "Build": 0,<br>  "Revision": 1,<br>  "MajorRevision": 0,<br>  "MinorRevision": 1<br>\} | 6.2.3 | \[<br>  \{<br>    "Major": 1,<br>    "Minor": 0,<br>    "Build": \-1,<br>    "Revision": \-1,<br>    "MajorRevision": \-1,<br>    "MinorRevision": \-1<br>  \},<br>  \{<br>    "Major": 2,<br>    "Minor": 0,<br>    "Build": \-1,<br>    "Revision": \-1,<br>    "MajorRevision": \-1,<br>    "MinorRevision": \-1<br>  \},<br>  \{<br>    "Major": 3,<br>    "Minor": 0,<br>    "Build": \-1,<br>    "Revision": \-1,<br>    "MajorRevision": \-1,<br>    "MinorRevision": \-1<br>  \},<br>  \{<br>    "Major": 4,<br>    "Minor": 0,<br>    "Build": \-1,<br>    "Revision": \-1,<br>    "MajorRevision": \-1,<br>    "MinorRevision": \-1<br>  \},<br>  \{<br>    "Major": 5,<br>    "Minor": 0,<br>    "Build": \-1,<br>    "Revision": \-1,<br>    "MajorRevision": \-1,<br>    "MinorRevision": \-1<br>  \},<br>  \{<br>    "Major": 5,<br>    "Minor": 1,<br>    "Build": 10032,<br>    "Revision": 0,<br>    "MajorRevision": 0,<br>    "MinorRevision": 0<br>  \},<br>  \{<br>    "Major": 6,<br>    "Minor": 2,<br>    "Build": 3,<br>    "Revision": \-1,<br>    "MajorRevision": \-1,<br>    "MinorRevision": \-1<br>  \}<br>\] | Linux 4.19.76\-linuxkit \#1 SMP Tue May 26 11:42:35 UTC 2020 | \{<br>  "Major": 3,<br>  "Minor": 0,<br>  "Build": \-1,<br>  "Revision": \-1,<br>  "MajorRevision": \-1,<br>  "MinorRevision": \-1<br>\} | \{<br>  "Major": 2,<br>  "Minor": 3,<br>  "Build": \-1,<br>  "Revision": \-1,<br>  "MajorRevision": \-1,<br>  "MinorRevision": \-1<br>\}


And with `Out-String` it looks like this:

| PSVersion | PSEdition | GitCommitId | OS | Platform | PSCompatibleVersions | PSRemotingProtocolVersion | SerializationVersion | WSManStackVersion
| --- | --- | --- | --- | --- | --- | --- | --- | ---
| Core | <br>Major  Minor  Build  Revision<br>\-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-\-\-\-<br>3      0      \-1     \-1<br><br> | Unix | <br>Major  Minor  Build  Revision<br>\-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-\-\-\-<br>2      3      \-1     \-1<br><br> | 6.2.3 | <br>Major  Minor  Build  Revision<br>\-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-\-\-\-<br>1      1      0      1<br><br> | Linux 4.19.76\-linuxkit \#1 SMP Tue May 26 11:42:35 UTC 2020 | <br>Major  Minor  Build  Revision<br>\-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-\-\-\-<br>1      0      \-1     \-1<br>2      0      \-1     \-1<br>3      0      \-1     \-1<br>4      0      \-1     \-1<br>5      0      \-1     \-1<br>5      1      10032  0<br>6      2      3      \-1<br><br> | <br>Major  Minor  Patch  PreReleaseLabel BuildLabel<br>\-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-  \-\-\-\-\-\-\-\-\-\-\-\-\-\-\- \-\-\-\-\-\-\-\-\-\-<br>6      2      3                      <br><br>


Which one you think is better?

**Comment by glicht** [comment]
I suggest using `ToString()` on each object property.

For example: 
image

**Comment by hod-alpert** [comment]
@glicht `Tostring` does not work well with arrays and hash tables
image

**Comment by hod-alpert** [comment]
@glicht Can you please re-review?

**Review by glicht — APPROVED** [review]
Excellent

**Pull request merged**